rfcs/text/0000-application-monitoring-for-ga.md

```diff
@@ -0,0 +1,146 @@
+### DCP PR:
+
+# System monitoring for General Availability
+
+## Summary
+
+This RFC describes the first standard revision for monitoring the distributed systems of the Human Cell Atlas (HCA) Data Coordination Platform (DCP). The standard is scoped to the minimum viable set of monitoring infrastructure the DCP will need to reach General Availability (GA). We anticipate that future revisions will be needed to meet our Service Level Objectives (SLOs) as DCP matures.
+
+The term monitoring is used here as defined in the [Google Site Reliability Engineering (SRE)](https://landing.google.com/sre/sre-book/chapters/monitoring-distributed-systems/) program. This document borrows heavily from SRE.
+
+## Author(s)
+
+* [Matthew Weiden](mailto:mweiden@chanzuckerberg.com)
+
+## Shepherd
+
+
+## Motivation
+
+Monitoring distributed systems [enables operators](https://landing.google.com/sre/sre-book/chapters/monitoring-distributed-systems/#why-monitor-pWsBTZ) to analyze long-term trends in system performance, compare experiments against a baseline, alert on failures, and conduct retrospective analysis on systems (debugging). These abilities are critical to keeping software correct and available.
+
+### User Stories
+
+* As a DCP operator, I want to compare experiments against a baseline, so that I can compare design alternatives.
```

Comment:
Who is a DCP operator? Are these team members from each component in the existing DCP deployment? Is this a member of a separate group in the existing deployment? Or is this an operator from an entirely separate group deploying another instance of the DCP? The answer will have a big impact on what one would consider necessary monitoring.

Reply:
It could mean either. Here I mean DCP team members operating deployments for the HCA.

Reply:
So would these DCP team members have an expert knowledge of the system or very good access to such experts? Or do we need to provide alerts that have a clear link to resolutions that do not require such expert knowledge? A suggestion: it could be that in a phased plan the DCP operator group is regarded as expert or expert supported and a later RFC addresses monitoring for non-experts?

Reply:
I wrote this from the "DevOps" perspective that the people who are operating the system would have been involved in its development and have some knowledge of its infrastructure. I believe this is aligned with the less extensive part of your suggestion.

Reply:
Could you explicit state that DCP operator here denotes people involved in the system development?